flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/reuse/ReplaceScanWithCalcShuttle.java

```diff
@@ -57,4 +61,12 @@ public RelNode visit(RelNode rel) {
 
         return super.visit(rel);
     }
+
+    private void visitDppSource(RelNode scan) {
+        // If scan is BatchPhysicalDynamicFilteringTableSourceScan,the input should be recursive
```

Comment:
nit:  If scan is `BatchPhysicalDynamicFilteringTableSourceScan`, traverse its input first